RadioController.py

```diff
@@ -101,157 +109,302 @@ def parse_data(self, type_id, byte_list, length):
+        curr_byte.next(math.floor(other_bit_field_length / 8))
+        return data
 
-    def message(self, byte_list, length):  # TODO
+    def message(self, byte_list, **kwargs):
         """
 
         :param byte_list:
         :type byte_list:
-        :param length:
-        :type length:
+        :param **kwargs:
+        :type **kwargs:
         :return:
         :rtype:
         """
-        converted = {0.0}  # STUB
-        return converted
+        data: Dict = {}
+
+        # Two step: int[] -> bytearray -> string. Probably a more efficient way
+        byte_data = bytearray(byte_list)
+        data[SubpacketEnum.MESSAGE.value] = byte_data.decode('ascii')
+        print(data[SubpacketEnum.MESSAGE.value])  # TODO Temporary until: saving to RocketData, logging, or displaying done
+        return data
 
-    def event(self, byte_list, length):  # TODO
+    def event(self, byte_list, **kwargs):
         """
 
         :param byte_list:
         :type byte_list:
-        :param length:
-        :type length:
+        :param **kwargs:
+        :type **kwargs:
         :return:
         :rtype:
         """
-        converted = {0.0}  # STUB
-        return converted
+        data: Dict = {}
+        # TODO Enumerate the list of events mapped to strings somewhere? Then convert to human readable string here.
+        data[SubpacketEnum.EVENT.value] = byte_list[0]
+        return data
 
-    def config(self, byte_list, length):  # TODO
+    # TODO. This is a stub until there is a spec.
+    def config(self, byte_list, **kwargs):
         """
 
         :param byte_list:
         :type byte_list:
-        :param length:
-        :type length:
+        :param **kwargs:
+        :type **kwargs:
         :return:
         :rtype:
         """
-        converted = {0.0}  # STUB
-        return converted
+        data = {}
+        data[SubpacketEnum.CONFIG.value] = byte_list[0]
+        return data
 
-    def single_sensor(self, byte_list, length):  # TODO
+    def single_sensor(self, byte_list, **kwargs):
         """
 
         :param byte_list:
         :type byte_list:
-        :param length:
-        :type length:
+        :param **kwargs:
+        :type **kwargs:
         :return:
         :rtype:
         """
-        converted = {0.0}  # STUB
-        return converted
+        type_id = kwargs['type_id']
+        data = {}
+        # TODO Commented out, since apparently we pass along state, or other non-floats as float too??
+        # if type_id == SubpacketEnum.STATE.value:
+        #     data[type_id] = byte_list[0]
+        # else:
+        #     data[type_id] = self.fourtofloat(byte_list[0])
+        data[type_id] = self.fourtofloat(byte_list[0])
+        return data
 
-    def gps(self, byte_list, length):  # TODO
+    def gps(self, byte_list, **kwargs):
         """
 
         :param byte_list:
         :type byte_list:
-        :param length:
-        :type length:
+        :param **kwargs:
+        :type **kwargs:
```

Comment:
Not entirely sure if this is how we're supposed to do documentation for *args and **kwargs. Might be worth double-checking. Same goes for other instances of this.

Reply:
Tried something out after searching, from https://stackoverflow.com/questions/47617701/python-how-to-docstring-kwargs-and-their-expected-types. Works nice in docstring format I think, and intellisense. Lmk if you prefer another way.

Reply:
No, that's fine by me. Just wanted to make sure we're not making up our own standard.